cli/azd/pkg/project/service_target_dotnet_containerapp.go

```diff
@@ -176,40 +189,134 @@ func (at *dotnetContainerAppTarget) Deploy(
+					task.SetError(err)
 					return
 				}
-				manifest = generatedManifest
+			}
+
+			task.SetProgress(NewServiceProgress("Fetching endpoints for container app service"))
+
+			containerAppTarget := environment.NewTargetResource(
+				targetResource.SubscriptionId(),
+				targetResource.ResourceGroupName(),
+				serviceConfig.Name,
+				string(infra.AzureResourceTypeContainerApp))
+
+			endpoints, err := at.Endpoints(ctx, serviceConfig, containerAppTarget)
+			if err != nil {
+				task.SetError(err)
+				return
+			}
+
+			task.SetResult(&ServiceDeployResult{
+				Package: packageOutput,
+				TargetResourceId: azure.ContainerAppRID(
+					targetResource.SubscriptionId(),
+					targetResource.ResourceGroupName(),
+					serviceConfig.Name,
+				),
+				Kind:      ContainerAppTarget,
+				Endpoints: endpoints,
+			})
+		},
+	)
+}
+
+func (at *dotnetContainerAppTarget) deployFromBicep(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	targetResource *environment.TargetResource,
+	remoteImageName string,
+	portNumber int,
+) error {
+	var modulePath string
+
+	appHostRoot := serviceConfig.DotNetContainerApp.AppHostPath
+	if f, err := os.Stat(appHostRoot); err == nil && !f.IsDir() {
+		appHostRoot = filepath.Dir(appHostRoot)
+	}
+
+	existingPath := filepath.Join(
+		appHostRoot, "infra", fmt.Sprintf("%s.bicep", serviceConfig.DotNetContainerApp.ProjectName))
+	if _, err := os.Stat(existingPath); err == nil {
+		log.Printf("using bicep module from %s", existingPath)
+		modulePath = existingPath
+	} else {
+		log.Printf(
+			"generating container app manifest from %s for project %s",
+			serviceConfig.DotNetContainerApp.AppHostPath,
+			serviceConfig.DotNetContainerApp.ProjectName)
+
+		path, dir, err := apphost.BicepModuleForProject(
+			serviceConfig.DotNetContainerApp.Manifest,
+			serviceConfig.DotNetContainerApp.ProjectName,
+			apphost.AppHostOptions{},
+		)
+		if dir != "" {
+			defer func() { _ = os.RemoveAll(dir) }()
+		}
+		if err != nil {
+			return fmt.Errorf("generating container app manifest: %w", err)
+		}
+		modulePath = path
+	}
+
+	compiledModule, err := at.bicepCli.Build(ctx, modulePath)
+	if err != nil {
+		return fmt.Errorf("compiling bicep module: %w", err)
+	}
+
+	deploymentName := fmt.Sprintf("%s-%d", serviceConfig.DotNetContainerApp.ProjectName, at.clock.Now().Unix())
+	projectParams, err := apphost.BicepParamsForProject(
+		serviceConfig.DotNetContainerApp.Manifest,
+		serviceConfig.DotNetContainerApp.ProjectName,
+		apphost.AppHostOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("getting bicep parameters: %w", err)
+	}
+
+	knownParams := map[string]any{
+		"location":                  at.env.Getenv("AZURE_LOCATION"),
+		"containerAppEnvironmentId": at.env.Getenv("AZURE_CONTAINER_APPS_ENVIRONMENT_ID"),
+		"containerRegistryEndpoint": at.env.Getenv("AZURE_CONTAINER_REGISTRY_ENDPOINT"),
+		"managedIdentityClientId":   at.env.Getenv("MANAGED_IDENTITY_CLIENT_ID"),
```

Comment:
Just wondering about this, does this "managed identity client id" have a more specific purpose? I'm wondering if this was more of an "uber role" that we may be able to wiggle out with `project.v1". @vhvb1989 in case he knows better.

Reply:
This is the client id of the managed identity that is used to run the container.  We need it to set `AZURE_CLIENT_ID` as an env-var for the app when we run it so that DAC will use the correct User Assigned identity at runtime.

Reply:
I see. So my mental model is that this would be described fully as  `SERVICE_<x>_MANAGED_IDENTITY_CLIENT_ID`. That doesn't sound scary at all, just a naming thing.